Update loadbalancing.md (#28991)
dotnet/AspNetCore.Docs: aspnetcore/grpc/loadbalancing.md

Addresses, connections and other load balancing state is stored in a `GrpcChannel` instance. A channel must be reused when making gRPC calls for load balancing to work correctly.

> [!IMPORTANT]

> > Advanced load balanacing configure uses dependency injection (DI). Apps that don't use DI can create a <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> instance. If an app already has DI setup, like an ASP.NET Core website, then types should be registered with the existing DI instance. `GrpcChannelOptions.ServiceProvider` is configured by getting an <xref:System.IServiceProvider> from DI.

Reviewer: ```suggestion
> Advanced load balancing configuration uses dependency injection (DI). Apps that don't use DI can create a <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> instance. If an app already has DI setup, like an ASP.NET Core website, then types should be registered with the existing DI instance. `GrpcChannelOptions.ServiceProvider` is configured by getting an <xref:System.IServiceProvider> from DI.
```

var response = await client.SayHelloAsync(new HelloRequest { Name = "world" });
```

## Use load balancing with gRPC client factory

> gRPC client factory can be configured to use load balancing. Configure clients with a load balancing address, specify channel credentials, and register any DI types required with the app's <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection>:

Author: ```suggestion
[gRPC client factory](xref:grpc/clientfactory) can be configured to use load balancing:
```

```csharp
var builder = WebApplication.CreateBuilder(args);

builder.Services.AddSingleton<ResolverFactory>(
    sp => new DnsResolverFactory(refreshInterval: TimeSpan.FromSeconds(30)));

builder.Services
    .AddGrpcClient<Greeter.GreeterClient>(o =>
    {
        o.Address = new Uri("dns:///my-example-host");
    })
    .ConfigureChannel(o => o.Credentials = ChannelCredentials.Insecure);

var app = builder.Build();
```

> 

Author: ```suggestion

The proceeding code:

* Configures the client with a load-balancing address.
* Specifies channel credentials.
* Registers required DI types with the app's <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection>.
```

var response = await client.SayHelloAsync(new HelloRequest { Name = "world" });
```

## Use load balancing with gRPC client factory

[gRPC client factory](xref:grpc/clientfactory) can be configured to use load balancing:

```csharp
var builder = WebApplication.CreateBuilder(args);

> builder.Services.AddSingleton<ResolverFactory>(
>     sp => new DnsResolverFactory(refreshInterval: TimeSpan.FromSeconds(30)));
> 
> builder.Services
>     .AddGrpcClient<Greeter.GreeterClient>(o =>
>     {
>         o.Address = new Uri("dns:///my-example-host");
>     })
>     .ConfigureChannel(o => o.Credentials = ChannelCredentials.Insecure);

Author: ```suggestion
builder.Services
    .AddGrpcClient<Greeter.GreeterClient>(o =>
    {
        o.Address = new Uri("dns:///my-example-host");
    })
    .ConfigureChannel(o => o.Credentials = ChannelCredentials.Insecure);

builder.Services.AddSingleton<ResolverFactory>(
    sp => new DnsResolverFactory(refreshInterval: TimeSpan.FromSeconds(30)));
```

var app = builder.Build();
```

The proceeding code:

* Configures the client with a load-balancing address.
* Specifies channel credentials.

> * Registers DI types with the app's <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection>.
> ## Write custom resolvers and load balancers

Author: ```suggestion
* Registers DI types with the app's <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection>.

## Write custom resolvers and load balancers
```

var response = await client.SayHelloAsync(new HelloRequest { Name = "world" });
```

## Use load balancing with gRPC client factory

[gRPC client factory](xref:grpc/clientfactory) can be configured to use load balancing:

```csharp
var builder = WebApplication.CreateBuilder(args);

builder.Services
    .AddGrpcClient<Greeter.GreeterClient>(o =>
    {
        o.Address = new Uri("dns:///my-example-host");
    })
    .ConfigureChannel(o => o.Credentials = ChannelCredentials.Insecure);

builder.Services.AddSingleton<ResolverFactory>(
    sp => new DnsResolverFactory(refreshInterval: TimeSpan.FromSeconds(30)));

var app = builder.Build();
```

> The proceeding code:

Reviewer: ```suggestion
The preceding code:
```
